### PR TITLE
Cosmos updates

### LINKED ISSUE #3555: Document FromSqlRaw for Cosmos
Also fix sample in https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-6.0/whatsnew#raw-sql-queries-with-fromsql
We don't append the discriminator condition anymore

### LINKED ISSUE #3730: Document Cosmos TTL configuration
See https://github.com/dotnet/efcore/issues/17307

### LINKED ISSUE #4017: Cosmos provider has many more limitations than those documented
Here are a few Cosmos Provider limitations that **should** be mentioned on this page:
- EF Core Migrations are not supported by the Cosmos Provider. Since this wasn't specifically mentioned on this page, I tried to use Migrations, and it ended...poorly. The error message returned by the Cosmos Provider when trying to initialize migrations is less than useless. Please save other developers the time and frustration by specifically mentioning this limitation on this page.
- Indexes defined on the DbContext are not created by EnsureDatabaseExists.  The containers are created, but the indexes are not.
- Check constraints, et al - I haven't specifically tried these yet, but I'm betting they don't work either.  If not, please let us know on this page.

I've spent most of my career working with Oracle/SQL Server databases, so my feedback is tainted by my RDBMS background. Nonetheless, I hope this was helpful.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a102d2e4-41d0-4619-a463-3e3862311a01
* Version Independent ID: a80b35f2-09dc-da67-df83-253ca5fd78cb
* Content: [Azure Cosmos DB Provider - Limitations - EF Core](https://docs.microsoft.com/en-us/ef/core/providers/cosmos/limitations)
* Content Source: [entity-framework/core/providers/cosmos/limitations.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/providers/cosmos/limitations.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **avickers**

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #4017
Fixes #3730
Fixes #3555

Part of #2325

### CODE REVIEW

File: entity-framework/core/providers/cosmos/functions.md [hunk 1]
```diff
@@ -11,82 +11,82 @@ This page shows which .NET members are translated into which SQL functions when
 
 ## Date and time functions
 
-.NET                  | SQL
---------------------- | ---
-DateTime.UtcNow       | GetCurrentDateTime()
-DateTimeOffset.UtcNow | GetCurrentDateTime()
+| .NET                  | SQL                  |
```
Comment (roji) [inline entity-framework/core/providers/cosmos/functions.md:14]: Any reason to have the closing vertical line on the right side of the table? Seems like needless extra management

File: entity-framework/core/providers/cosmos/index.md [hunk 2]
```diff
@@ -57,14 +57,88 @@ Saving and querying data follows the normal EF pattern:
 > [!IMPORTANT]
 > Calling [EnsureCreatedAsync](/dotnet/api/Microsoft.EntityFrameworkCore.Storage.IDatabaseCreator.EnsureCreatedAsync) is necessary to create the required containers and insert the [seed data](xref:core/modeling/data-seeding) if present in the model. However `EnsureCreatedAsync` should only be called during deployment, not normal operation, as it may cause performance issues.
 
+## Connecting and authenticating
+
+The Azure Cosmos DB provider for EF Core has multiple overloads of the [UseCosmos](/dotnet/api/Microsoft.EntityFrameworkCore.CosmosDbContextOptionsExtensions.UseCosmos) method. These overloads support the different ways that a connection can be made to the database, and the different ways of ensuring that the connection is secure.
+
+> [!IMPORTANT]
+> Make sure to understand [_Secure access to data in Azure Cosmos DB_](/azure/cosmos-db/secure-access-to-data) to understand the security implications and best practices for using each overload of the `UseCosmos` method.
```
Comment (roji) [inline entity-framework/core/providers/cosmos/index.md:65]:
```suggestion
> Make sure to read [_Secure access to data in Azure Cosmos DB_](/azure/cosmos-db/secure-access-to-data) to understand the security implications and best practices for using each overload of the `UseCosmos` method.
```

File: entity-framework/core/providers/cosmos/limitations.md [hunk 3]
```diff
@@ -2,29 +2,27 @@
...
+The Azure Cosmos DB database provider targets the Azure Cosmos DB NoSQL store, which is a [document database](https://en.wikipedia.org/wiki/Document-oriented_database). Most EF Core providers target [relational databases](https://en.wikipedia.org/wiki/Relational_database). Document databases and relational databases behave in fundamentally different ways. EF Core does not attempt to hide these differences; rather EF Core provides common patterns that can be used successfully across both kinds of database, together with features tailed to a particular provider that follow best practices for a given type of database. If a feature of EF Core is a pit-of-failure for a certain kind of database, then typically the database provider will not implement that feature, and instead help steer uses towards a [pit-of-success](/archive/blogs/brada/the-pit-of-success) approach.
```
Comment (roji) [inline entity-framework/core/providers/cosmos/limitations.md:10]:
```suggestion
The Azure Cosmos database provider targets the Azure Cosmos DB NoSQL store, which is a [document database](https://en.wikipedia.org/wiki/Document-oriented_database). Most EF Core providers target [relational databases](https://en.wikipedia.org/wiki/Relational_database). Document databases and relational databases behave in fundamentally different ways. EF Core does not attempt to hide these differences; rather EF Core provides common patterns that can be used successfully across both kinds of database, together with features tailored to each specific database, following best practices there. If a feature of EF Core is a pit-of-failure for a certain kind of database, then typically the database provider does not implement that feature, and instead help steer uses towards a [pit-of-success](/archive/blogs/brada/the-pit-of-success) approach.
```

File: entity-framework/core/providers/cosmos/limitations.md [hunk 4]
```diff
@@ -2,29 +2,27 @@
...
+- Schema concepts defined on the EF model, like indexes and constraints, are ignored when using a document database, since there is no schema. Note that Azure Cosmos DB NoSQL performs [automatic indexing of documents](/azure/cosmos-db/index-overview).
+- Loading graphs of related entities from different documents is not supported. Document databases are not designed to perform joins across many documents; doing so would be very inefficient. Instead, it is more common to denormalize data so that everything needed is in one, or a small number, of documents. However, there are some forms of cross-document relationships that could be handled--see [Limited Include support for Cosmos](https://github.com/dotnet/efcore/issues/16920#issuecomment-989721078).
 
 ## Azure Cosmos DB SDK limitations
 
-- Only async methods are provided
+Some functionality supported by most EF Core database providers are not supported by the Azure Cosmos DB SDK. This includes synchronous APIs.
```
Comment (roji) [inline entity-framework/core/providers/cosmos/limitations.md:21]:
```suggestion
Some functionality supported by most EF Core database providers isn't supported by the Azure Cosmos DB SDK. This includes synchronous APIs.
```

Comment (roji) [inline entity-framework/core/providers/cosmos/limitations.md:21]: Not sure why we have this separate category of limitations here - I don't think it's very relevant which limitations come from the EF provider (e.g. no migrations) and which come from the underlying Cosmos SDK (no async). Should we just merge them?

**Review by roji — APPROVED** [review]

File: entity-framework/core/providers/cosmos/functions.md (hunk 1 above)
Comment (ajcvickers) [reply entity-framework/core/providers/cosmos/functions.md:14]: It's the way Rider formats tables, and makes them easier to read. Shouldn't be any extra work if you're using Rider, and I presume and decent IDE/editor has similar functionality.

File: entity-framework/core/providers/cosmos/limitations.md (hunk 4 above)
Comment (ajcvickers) [reply entity-framework/core/providers/cosmos/limitations.md]: @AndriySvyryd?

Comment (AndriySvyryd) [reply entity-framework/core/providers/cosmos/limitations.md]: It made sense previously, when the provider limitations were temporary. Now they can be merged

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**